command/certificate/remote.go

```diff
@@ -57,19 +64,26 @@ func getPeerCertificates(addr, serverName, roots string, insecure bool) ([]*x509
 // empty string (and 'isURL:false').
 //
 // Examples:
-// trimURL("https://smallstep.com/onbaording") -> "smallstep.com", true, nil
+// trimURL("https://smallstep.com/onbaording") -> "smallstep.com:443", true, nil
 // trimURL("https://ca.smallSTEP.com:8080") -> "ca.smallSTEP.com:8080", true, nil
 // trimURL("./certs/root_ca.crt") -> "", false, nil
-// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm", true, nil
+// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm:443", true, nil
 // trimURL("hTtPs://sMaLlStEp.cOm hello") -> "", false, err{"invalid url"}
 func trimURL(ref string) (string, bool, error) {
 	tmp := strings.ToLower(ref)
-	for _, prefix := range urlPrefixes {
+	for prefix, _ := range urlPrefixes {
 		if strings.HasPrefix(tmp, prefix) {
 			u, err := url.Parse(ref)
 			if err != nil {
 				return "", false, errors.Wrapf(err, "error parsing URL '%s'", ref)
 			}
+			_, port, _ := net.SplitHostPort(u.Host)
+			if port == "" {
+				p := uint64(urlPrefixes[prefix])
+				if p != 0 {
```

Comment:
Have you tested the behavior with unpassed port number? Does it reliably error put? If not, we should error check this one, or set the defaults of those to `443`.

Reply:
By default unpassed port are set to 443

https://github.com/smallstep/cli/blob/5c1724af8b4d5a0709e2917db89bb8a11966a7c1/command/certificate/remote.go#L39

I just set the default to 443 in the map, it make more sense.